Resolves issue #114: Update mod doc repo example and instructions

sterobin opened the pull request:
All, this address issue #114 and also expands the general repo guidance for first-timers, in conjunction with the mod doc repo workshop. Please review and provide any feedback. 

Here's the summary of changes and doc preview links:

**Summary of changes:**
* Replaced the former `simple-modular-guide.zip` with a fuller and more realistic `mod-doc-repo-example.zip`. Buildable in asciidoctor or (b)ccutil.
* Added new section on [Modular Documentation Repositories](http://file.rdu.redhat.com/~sterobin/modular-docs-manual/modular-docs-manual/master.html#modular-documentation-repositories) (preview requires VPN).
* Added disclaimer note about evolving recommendations in the [introduction section](http://file.rdu.redhat.com/~sterobin/modular-docs-manual/modular-docs-manual/master.html#introduction), which I reviewed with our tooling and writer subgroup.
* Restored missing components to enable the mod doc ref guide to be built with (b)ccutil for Pantheon, if needed (master-docinfo.xml, .gitignore for build folder, and missing context definition that disabled building).

ncbaratta commented:
lgtm

sterobin commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+This repository structure poses the following potential disadvantages:
+
+* *Extensive use of symlinks.* In order to enable titles to be built with distinct assemblies, modules, and images, this repository structure uses many symlinks (symbolic links) for access across directories. You can extract these symlinks with a build script or additional variable configurations (such as `imagesdir`), but because this example repository is intended for a general CCS audience, it ...
```
Good point @benradey. There's also another camp that is anti-symlinks as a whole, because they don't always work the same on every OS. Among us on CCS it's fairly safe, but on a community scale for example, most devs won't have it. So I'm hesitant to go into it here, but agree that you're right, a dedicated section with some guidance on symlinks would be good. I'll think on it and when I have some ideas, I'll create an issue for it.

sterobin commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+modules:
+  - modules/**/*.adoc
+resources:
+  - _images/**/*.png
```
Who/where would we need to go to for verification? Would love to confirm as part of this PR.

benradey replied:
Well I just tested it and it appears that it is not supported, unfortunately.

Assuming that this is a feature that CCS would want to have, would you mind creating a Jira ticket for it and then we can work with Dawn to prioritize it?

sterobin replied:
Per gchat with @benradey: Updated PR for now to be accurate and created CCSOPS ticket to request support for `/**/` notation for any number of sub-folders.

oraNod commented on modular-docs-manual/content/topics/introduction.adoc:
```diff
@@ -2,3 +2,5 @@
 
 This manual provides instructions on how to author modularly structured documentation based on user stories. The manual defines used terminology, describes components that form modular documentation, and instructs writers on how to use provided templates to turn user stories into modular documentation.
+
+IMPORTANT: As CCS strategic imperatives and Pantheon 2 development continue to evolve, the details and recommendations in this guide might be modified. The Flexible Customer Content (FCC) team recognizes that the fluctuation in recommendations can be frustrating for CCS teams trying to adopt modularization standards and apologizes in advance as the initiative pursues the best possible outcome for ...
```
Admonitions take the following format:

```
[IMPORTANT]
====
Read this note.
====
```

sterobin replied:
@oraNod That's just for mult-line block admonitions. For single-line admonitions, both formats render the same, so it's a matter of preference, imo. I prefer using this format for single-line admonitions because it's only one ascii line, faster, more streamlined. I reserve the other format for multi-line block notes, which is I believe why that format exists anyway: https://asciidoctor.org/docs/user-manual/#admonition

oraNod replied:
Fair enough. Just a review reflex on my part. I typically enforce the multi-line format throughout docs just to make it easy and so we're always consistent.

oraNod commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+* `assemblies`: Directory with assembly files. You collect and compile assemblies into documents (titles) using the `titles-enterprise` folder.
+* `modules`: Directory with module files, optionally divided into sub-directories by subject matter.
+* `titles-enterprise`: Directory with folders for each enterprise document or _title_ to be built in Pantheon. Each document folder contains a `master.adoc` file where you include one or more relevant assemblies that are built as part of the document. The `master-docinfo.xml` file contains metadata that is required by `bccutil` (and `ccutil`) when converting to DocBook format. (This metadata file ...
+* `pantheon2.yml`: YAML file with metadata that is required for publishing content on Pantheon 2:
```
No need to say "Directory"/"YAML file" imo. That should be obvs to users. Start with third person singular verbs.

* `_artifacts`: Contains attributes and metadata. Optionally prefix the name with an underscore character to denote a directory that does not contain content.
* `_images`: Contains image files that you can optionally organize into sub-directories. ... and so on.

"In cases where content exists in community repositories, you can create a `titles-community` directory that holds upstream only content."

sterobin replied:
I'm listing things that are all at the same top level of the repo, most of which directories, some files (like the YAML and README). There's really not a directory to specify since it depends on the user and not shown in the structure. I could just use an example parent dir name, but then should add the same to all the other directories listed for consistency (example/titles-enterprise). Then just noise to me. Seems unnecessary imo.

Yeah, would be fine to role with verb-first, but because some are folders and the one is a file, I wanted to start with what it is (directory or file), otherwise when I get to the YAML, it will also say "Contains ..." and could be odd, even with the parent dire there named. Not trying to split hairs, just keep it simple and clear. I think people get it, but that's just me. I'm open if you have a strong feeling otherwise.

Regarding that community rewording, I don't think that's quite accurate. The whole purpose of a single-sourced repo is to have mostly or all shared content, and then the community folder dedicated to the configs to just the community rendering, which is really all that's different. For content that's community only, that's controlled by including it in the community folder rendering files and not including in the enterprise rendering files. But again, this may vary depending on circumstances, so the description I went with is for this specific repo scenario, which would follow what I've explained here.

oraNod replied:
YAML files can contain metadata.

Sure thing with that community rewording. I just see passive and my instinct is to re-word.

sterobin replied:
Understood.

oraNod commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+This repository structure offers the following potential advantages:
+
+* *Flexibility in determining publishable titles.* In this structure, the images, artifacts, assemblies, and modules all serve as distinct building blocks that you use to compile a document or _title_. You can include an image in multiple modules, a module in multiple assemblies, and an assembly in multiple titles. Additionally, just as an assembly can include multiple modules, a title can contai ...
```
```suggestion
* *Flexibility in determining publishable titles.* In this structure, the images, artifacts, assemblies, and modules all serve as distinct building blocks that you use to compile a document or _title_. You can include an image in multiple modules, a module in multiple assemblies, and an assembly in multiple titles. Additionally, just as assemblies can include multiple modules, titles can contain multiple assemblies.
```

sterobin replied:
Plural "assemblies" containing multiple modules and "titles" containing multiple assemblies doesn't give the emphasis of multiple in the same assembly/title that we're trying to convey. Two titles each with one assembly still = plural titles with multiple assemblies. I was trying to emphasize multiple in one.

Boy, I'm sure being stiff on this PR so far but really not trying to :( I can give up and go with the flow if I need to. :)

oraNod replied:
2 in 1 shampoo never made sense to me. 1 isn't big enough to hold 2. That's why there's 2.

Jokes aside, it's fine if you don't want to make the change. I generally prefer using plural forms for consistency. Some day I'll have to explain the reason behind that but I can't put it into words at the moment.

sterobin replied:
Haha! Good one :)  Well I'll look forward to that rationale (honestly, not being sarcastic at all). Sounds interesting. For now, will leave as is, if okay with you.

oraNod commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+* *Flexibility in determining publishable titles.* In this structure, the images, artifacts, assemblies, and modules all serve as distinct building blocks that you use to compile a document or _title_. You can include an image in multiple modules, a module in multiple assemblies, and an assembly in multiple titles. Additionally, just as an assembly can include multiple modules, a title can contai ...
+* *More intuitive content reuse.* When you decouple the standalone content components from the publishable titles, you avoid potentially awkward includes when modules or assemblies from one title are reused in another. For example, consider a repository where each publishable title folder contains all modules or assemblies or both directly within it. This creates a dependency for that module or a ...
+* *Potentially easier collaboration in community and between content repositories.* For repositories that are single sourced with community projects or that need to reuse content between product documentation repositories, this structure enables you to isolate the enterprise title compilation requirements so that the standalone components can be shared more readily. However, single sourcing betwe ...
```
s/single sourcing/single-sourcing

sterobin replied:
Yeah, I've also said "single-sourcing", but a growing counter argument is that it's only hyphenated because it's a clipped form of the full phrase "single-sourced publishing/content/whatever". But you're right, definitely becoming the norm and very logical. Changed throughout.

oraNod commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+* *Potentially easier collaboration in community and between content repositories.* For repositories that are single sourced with community projects or that need to reuse content between product documentation repositories, this structure enables you to isolate the enterprise title compilation requirements so that the standalone components can be shared more readily. However, single sourcing betwe ...
+* *Consistency with Pantheon 2 YAML configurations.* In this structure, the division of modules, assemblies, and other resources is consistent with the division of components in the `pantheon2.yml` file that is required for documents published on Pantheon 2.
+
+This repository structure poses the following potential disadvantages:
```
5 counts of the word "potential" here. I'd consider revising to drop/reword some of those instances to avoid overuse.

sterobin replied:
Yeah, guilty :( I noticed it too, but I'm just so hesitant to be objective in some of these because of how many factors are at play. I ditched it from the two intro sentences before the lists. Left the other couple.

oraNod commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+This repository structure poses the following potential disadvantages:
+
+* *Extensive use of symlinks.* In order to enable titles to be built with distinct assemblies, modules, and images, this repository structure uses many symlinks (symbolic links) for access across directories. You can extract these symlinks with a build script or additional variable configurations (such as `imagesdir`), but because this example repository is intended for a general CCS audience, it ...
```
While we're talking about symlinks, is it worth mentioning that Pantheon does not support `../` and requires symlink instead? 

It might be overkill for this doc but in some community projects symlinks are verboten. You can skirt around it with variables. 

For example, consider an includes directive like this: `includes::{modules}/proc_doing_something.adoc` 

In the title files you can then have one of the following:

Product/enterprise: `:modules: modules` (where `modules` is the name of the symlink)
Community: `:modules: ../path/to/modules`

sterobin replied:
Definitely. @benradey also brought it up. His suggestion was for a section dedicated to considerations with symlinks. I agree with that but not up for it as part of this PR :)

oraNod replied:
Sounds good.

oraNod commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+[source]
+----
+$ ccutil compile --lang=en-US --format html-single --open
+----
```
If ccutil is deprecated, let's not mention it if this is for newbies.

oraNod replied:
Or maybe move ccutil below bccutil and asciidoctor.

sterobin replied:
You're right. I shouldn't say "deprecated". I was using the tooling language and they consider it deprecated. It just applies to RHEL7 and earlier. I still use it for example, on RHEL 7 CSB. I've reworded to just say "(RHEL 7 or Fedora 29 and earlier)" to match the bccutil snippet. I swapped the ordering though as you said to go from most recent/desired to least. Good call.

rkratky commented on modular-docs-manual/content/topics/introduction.adoc:
```diff
@@ -2,3 +2,5 @@
 
 This manual provides instructions on how to author modularly structured documentation based on user stories. The manual defines used terminology, describes components that form modular documentation, and instructs writers on how to use provided templates to turn user stories into modular documentation.
+
+IMPORTANT: As CCS strategic imperatives and Pantheon 2 development continue to evolve, the details and recommendations in this guide might be modified. The Flexible Customer Content (FCC) team recognizes that the fluctuation in recommendations can be frustrating for CCS teams trying to adopt modularization standards and apologizes in advance as the initiative pursues the best possible outcome for ...
```
See my comment about references to internal tooling (applies to teams as well): https://github.com/redhat-documentation/modular-docs/pull/121#discussion_r449559189

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+[id="modular-documentation-repositories"]
+= Modular Documentation Repositories
+
+IMPORTANT: As CCS strategic imperatives and Pantheon 2 development continue to evolve, the details and recommendations in this section and possibly other sections in this guide might be modified. The Flexible Customer Content (FCC) team recognizes that the fluctuation in recommendations can be frustrating for CCS teams trying to adopt modularization standards and apologizes in advance as the init ...
```
Again... internal RH reference. See https://github.com/redhat-documentation/modular-docs/pull/121#discussion_r449559189

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+IMPORTANT: As CCS strategic imperatives and Pantheon 2 development continue to evolve, the details and recommendations in this section and possibly other sections in this guide might be modified. The Flexible Customer Content (FCC) team recognizes that the fluctuation in recommendations can be frustrating for CCS teams trying to adopt modularization standards and apologizes in advance as the init ...
+
+Due to many factors that determine documentation repository design for individual Red Hat products, the Flexible Customer Content (FCC) initiative cannot provide a strict documentation repository template for all CCS teams to follow. Each team faces different implementation challenges, especially when Red Hat product documentation is single-sourced with corresponding community project documentati ...
```
I suggest rewording to avoid mentions of FCC, CCS, etc. This paragraph is applicable to upstream and other organizations as well.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+Due to many factors that determine documentation repository design for individual Red Hat products, the Flexible Customer Content (FCC) initiative cannot provide a strict documentation repository template for all CCS teams to follow. Each team faces different implementation challenges, especially when Red Hat product documentation is single-sourced with corresponding community project documentati ...
+
+However, for optimal FCC, Pantheon, and Pantheon 2 compliance, the FCC initiative recommends a general repository structure similar to the following example, where possible:
```
I suggest rewording to avoid mentions of FCC, CCS, Pantheon, etc. This paragraph is applicable to upstream and other organizations as well.

rkratky replied:
In fact, I'd remove this paragraph altogether. This is a note that could be added to every single recommendation in these guidelines. As such, it serves no purpose here.

sterobin replied:
I've added the note at the top of the entire doc, in the intro, but it started here because of emphasis. This issue has been particularly problematic with repo recommendations, so I just wanted to resurface it for clarity and great visibility. Regarding internal references, responding to that separately.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+    ├── _artifacts -> ../_artifacts  // Symlinks can be extracted.
+    ├── _images -> ../_images
+    ├── modules -> ../modules
+    ├── assembly-kogito-configuring.adoc
```
Let's use the underscore to separate prefixes. It's not a hard requirements, and hyphens are OK, but underscores are more commonly used and more practical from the mod docs perspective. So, for examples, I think we should go with them. (Regardless of what is selected, all examples in the guidelines need to be consistent.)

sterobin replied:
That would mean a very large change to the entire repo, which uses hyphens. We've decided as a team, because there's so much discord over this, that we officially can't really lean to one or the other anymore, and both are viable options. So I'd suggest keeping this in the example, since whichever route we go, we'll likewise have someone coming in to argue for the other.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+However, for optimal FCC, Pantheon, and Pantheon 2 compliance, the FCC initiative recommends a general repository structure similar to the following example, where possible:
+
+.Recommended documentation repository structure (where possible)
```
I think this example should not copy an existing repo. Actual assembly and module names (instead of generics, such as `proc_procedure-module.adoc`) invite confusion.

sterobin replied:
I've received overwhelming feedback for exactly the opposite in my mod doc repo trainings and other conversations with various teams. Having concrete, realistic, and various examples was and has been a huge ask. I got a lot of complaints for the former generic and simplified repo example.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+        ├── _images -> ../../_images
+        ├── assemblies -> ../../assemblies
+        ├── master.adoc
+        └── master-docinfo.xml
```
Docinfo files are specific to ccutil and Red Hat. I don't think they should be included in this.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+├── _artifacts
+├── _images
+├── assemblies
+    ├── _artifacts -> ../_artifacts  // Symlinks can be extracted.
```
What does 'extracted' mean in this context?

sterobin replied:
I explain in the bulletted description that follows. Can do so with variables, build tooling, scripting, whatever. I was trying to avoid a long callout here, and thus provided more detail in the description that follows.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+└── README.md
+----
+
+This repository structure contains the following key components:
```
I'd suggest using callouts for these descriptions.

sterobin replied:
Haha, funny you say that. There has been explicit guidance from the style team to avoid callouts where possible, for accessibility reasons and for minimizing clutter. So I'm afraid I'm stuck between two conflicting opinions on this.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+* `_images`: Directory with image files, optionally divided into sub-directories by subject matter. The underscore in the folder name is an optional indication of a non-content folder.
+* `assemblies`: Directory with assembly files. You collect and compile assemblies into documents (titles) using the `titles-enterprise` folder.
+* `modules`: Directory with module files, optionally divided into sub-directories by subject matter.
+* `titles-enterprise`: Directory with folders for each enterprise document or _title_ to be built in Pantheon. Each document folder contains a `master.adoc` file where you include one or more relevant assemblies that are built as part of the document. The `master-docinfo.xml` file contains metadata that is required by `bccutil` (and `ccutil`) when converting to DocBook format. (This metadata file ...
```
Let's not refer to internal RH tooling.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+* `assemblies`: Directory with assembly files. You collect and compile assemblies into documents (titles) using the `titles-enterprise` folder.
+* `modules`: Directory with module files, optionally divided into sub-directories by subject matter.
+* `titles-enterprise`: Directory with folders for each enterprise document or _title_ to be built in Pantheon. Each document folder contains a `master.adoc` file where you include one or more relevant assemblies that are built as part of the document. The `master-docinfo.xml` file contains metadata that is required by `bccutil` (and `ccutil`) when converting to DocBook format. (This metadata file ...
+* `pantheon2.yml`: YAML file with metadata that is required for publishing content on Pantheon 2:
```
This is RH internal (unfortunately, despite promises to the contrary, Pantheon is not being developed as an open(-source) project. Therefore, references to the tool should be kept out of these guidelines, so as not to detract from their value for other organizations and upstream projects.

If we need internal RH guidelines for publishing, they should be kept separately from this content.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+    ├── kogito-developing-decision-services  // Contains multiple assemblies.
+        ├── ...
+    ├── ...
+├── pantheon2.yml
```
Specific to RH.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+This repository structure poses the following disadvantages:
+
+* *Extensive use of symlinks.* In order to enable titles to be built with distinct assemblies, modules, and images, this repository structure uses many symlinks (symbolic links) for access across directories. You can extract these symlinks with a build script or additional variable configurations (such as `imagesdir`), but because this example repository is intended for a general CCS audience, it ...
```
```suggestion
* *Extensive use of symlinks.* In order to enable titles to be built with distinct assemblies, modules, and images, this repository structure uses many symlinks (symbolic links) for access across directories. You can eliminate these symlinks with a build script or additional variable configurations (such as `imagesdir`), but because this example repository is intended for a general audience, it maintains explicit symlinks, so that a user can build the repository titles with standard Asciidoctor tooling. Alternative repository structures that combine all relevant images, modules, and assemblies together in title folders can reduce the number of symlinks, but these alternative structures then forego many of the advantages listed previously.
```

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+This repository structure poses the following disadvantages:
+
+* *Extensive use of symlinks.* In order to enable titles to be built with distinct assemblies, modules, and images, this repository structure uses many symlinks (symbolic links) for access across directories. You can extract these symlinks with a build script or additional variable configurations (such as `imagesdir`), but because this example repository is intended for a general CCS audience, it ...
+* *Potential conflicts with community content strategy.* Although this documentation repository structure is typically more collaboration-friendly than other FCC repository alternatives (as described in the previous advantages), the FCC strategy as a whole still might not be well received among all community projects. For example, many community projects prefer Mavenized repositories divided into ...
```
Redundant references to (internal) FCC.

The whole paragraph is a speculation. I don't think it needs to be here.

sterobin replied:
Right, understood. I explained to the group in the call about this that a recurring complaint I keep receiving in my courses and work with other teams is that the mod doc ref guide is great at explaining what to do, but not really why. Teams with differing strategies and opinions are eager for the rationale in some things, particularly with repo design. So in my initial research and planning for this repo example and documentation, one big request was to please explain the advantages and disadvantages of the general strategy here, since it's up to individual teams to assess and decided, based on their many factors. So some of this is indeed speculation and potential, but that's what I've been asked by many to provide. So I'd prefer to keep it as an answer to some of the potential why's.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,141 @@
+
+To build the example titles, in a command terminal, navigate to `~/titles-enterprise/__DOC_FOLDER__` and enter one of the following commands:
+
+.Build with `bccutil` (RHEL 8 or Fedora 30 and later)
+[source]
+----
+$ bccutil compile --lang=en-US --format html-single --open
+----
+
+.Build with `ccutil` (RHEL 7 or Fedora 29 and earlier)
+[source]
+----
+$ ccutil compile --lang=en-US --format html-single --open
+----
```
Internal references. This tooling is not available publicly.

sterobin commented:
@rkratky Thanks so much! I committed your re-wording suggestion, and replied on the rest.

Regarding the internal reference issue. I've left as is for now because this raises something that I think we need to bring up on the next team call. There are many important internal suggestions and recommendations that we need to make clear, so we either need to add some attributes and then have different build options for community vs. enterprise, or agree that internal references are okay. I'm fine either way. Just a heads-up, however, that in my workings and research on single-sourced team repos, I've not yet found a team who's community project uses or even tolerates modularization and its conventions. There are growing escalations over this among many projects, and I'm involved with the ascii splitter work that is being taken very seriously now by communities. They seem to be looking for any way out of modularization, from what I can see.

Anyway, point is, I just want to make sure the efforts to avoid internal references are even worth it still. If they are, then great. But we need then to have some attributes and dual building options for this guide. Most or all of the internal refs in this PR for example are critical, imo. Same in #121 .

emmurphy1 commented:
lgtm

sterobin commented:
@rkratky , @emmurphy1 @Preeticp I've updated this PR to take the approach we discussed in today's meeting, which is to add the mod repo example in a separate [mod-doc-repo-example](https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example) branch, which I've just pushed upstream, and then simplify/broaden the mod repo section of the doc and link to that example branch for more info. I've also tweaked that blanket disclaimer that I added to the intro to not refer to internal terminology.

Let me know what you think and I'll move forward.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,6 @@
+[id="modular-documentation-repositories"]
+= Modular Documentation Repositories
+
+Due to many factors that determine documentation repository design for individual products, the modular documentation initiative cannot provide a strict documentation repository template for all documentation teams to follow. Each team faces different implementation challenges, especially when Red Hat enterprise product documentation is single-sourced with corresponding community project document ...
```
Keep it simple?
```suggestion
Due to many factors that determine repository design for individual products, these modular documentation guidelines do not provide a strict documentation repository template.
```

sterobin replied:
Normally I'd agree, but I want to avoid being so concise and vague here. I want it crystal clear why we can't provide the strict template. Just saying "due to many factors" alone isn't sufficient, imo. So I'd prefer to keep the wording as is.

rkratky replied:
I don't like "due to many factors", too, and wanted to cut even that...

My point is that it's a manual. Manuals should instruct while being as concise as possible -- not elaborate on reasons regarding the inclusion or exclusion of this or that. If you feel so strongly about it, I suggest we create an FAQ section at the end.

sterobin replied:
Okay @rkratky , I give up :) For the sake of getting this through, I've taking your section here and below. Done.

rkratky commented on modular-docs-manual/content/topics/module_modular-documentation-repositories.adoc:
```diff
@@ -0,0 +1,6 @@
+
+Due to many factors that determine documentation repository design for individual products, the modular documentation initiative cannot provide a strict documentation repository template for all documentation teams to follow. Each team faces different implementation challenges, especially when Red Hat enterprise product documentation is single-sourced with corresponding community project document ...
+
+However, for a general repository structure that the modular documentation initiative recommends for optimal results with modular standards, see the example documentation repository and `README` file in the https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example[`mod-doc-repo-example`] branch in this repository.
```
```suggestion
For a general repository structure used by documentation teams at Red Hat, see the example documentation repository and `README` file in the https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example[`mod-doc-repo-example`] branch in this repository.
```

sterobin replied:
Again, wanting to be clear, but also, this rewording is necessarily correct. It's not a repo design "used by doc teams at Red Hat", at least not yet. Only a few teams are somewhat similar to this. Otherwise, teams are all over the map. That's kind of the purpose of the repo example, to give a recommended structure for optimal results with mod doc. So I guess I again would prefer the existing wording over the rewording.

rkratky replied:
* I want to make it clear that the example is RH-specific -- as discussed. We can qualify the statement ("some teams...").
* I I think it's superfluous to include any wording referring to a "modular docs initiative". This entire manual is a product of some initiative. In product docs, we don't say "The Customer Content Services team recommends..." :)
```suggestion
For a general repository structure used by some documentation teams at Red Hat, see the example documentation repository and `README` file in the https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example[`mod-doc-repo-example`] branch in this repository.
```

rkratky commented on modular-docs-manual/content/topics/introduction.adoc:
```diff
@@ -2,3 +2,5 @@
 
 This manual provides instructions on how to author modularly structured documentation based on user stories. The manual defines used terminology, describes components that form modular documentation, and instructs writers on how to use provided templates to turn user stories into modular documentation.
+
+IMPORTANT: As modular documentation imperatives continue to evolve, the details and recommendations in this guide might be modified. The modular documentation team recognizes that the fluctuation in recommendations can be frustrating for contributors trying to adopt modularization standards and apologizes in advance as the initiative pursues the best possible outcome for modularized documentation ...
```
Do we really need this? I think it's a given.

sterobin replied:
I'd say it's definitely not a given, based on the comments in the repo "Issues" and the consistent feedback on the mod doc manual that I get in my repo workshops. People are very agitated on the moving target and changes in file names, heading formatting modifications, template tweaks, etc. I have to say something like this note in reply each time it comes up, so we very much need this upfront clear disclaimer to address it, imo.

rkratky replied:
Understood. But then we're failing in educating ppl about the fact that it's a project like any other. It's even on GH. It's not set in stone. That doesn't invalidate your point, but I think we should be clear about that up front. All guidelines evolve. Even style guides are published in new revisions.

Could we perhaps cut it to just the first sentence? The rest is kind of a "don't-hate-us" apology, which, I think, shouldn't be a part of docs.

sterobin replied:
Kept first sentence

rkratky commented on modular-docs-manual/master-docinfo.xml:
```diff
@@ -0,0 +1,8 @@
+<productname>FCC</productname>
```
Why is this file included?

sterobin replied:
This file and its components are required in order to build the ccutil version of the guide. It was half way set up to be able to do so, so I just fixed the rest to enable it, for publishing on Pantheon if we need to. We can make the `productname` whatever, but needs to be something there in order to build. So I chose FCC. Since it would only be used internally anyway, maybe doesn't matter.

rkratky replied:
This doc cannot be published through Pantheon because it's on GH, not internal GL. If anyone feels the need to build it through Pantheon, they will still need to set it up on GL where all RH-specific prereqs can be satisfied.

(Again, I believe we should keep this project as much org-agnostic as possible. Otherwise we're sending a msg of "This is a RH thing, and it's not suitable for wider adoption." RH has many upstream (docs) projects that get periodically downstreamed for publishing internally. None of those projects put downstream docinfo files into their upstream community repos.)

sterobin replied:
Done. Removed.

Preeticp commented:
LGTM
